### DISCUSSION TITLE
Adding New Functionality to This Library

### DISCUSSION THREAD

**Discussion by FA5I** [saleweaver/python-amazon-sp-api#98] [Q&A]
Hi,

Have been playing with the library and love what has been built so far.

Had a couple of questions:

1. Lets say I want to add some functionality for [FBA Small and Light](https://github.com/amzn/selling-partner-api-docs/blob/main/references/fba-small-and-light-api/fbaSmallandLight.md#selling-partner-api-for-fba-small-and-light). 

Would the best way to do this simply be to create the class directly in the ```/python-amazon-sp-api/sp_api/api``` folder?

2. The reason I ask is because I know that Codegen can be used to build the client SDK based on the Amazon documentation. From this generated client SDK, it seems that the code was edited to create this Python-amazon-sp-api library. 

So do I need to do some sort of Codegen to build the models, before I can start adding functionality? Apologies if this is a basic question but the swagger documentation was really poor and I have no experience with open API tools before.

Thanks!

**Comment by saleweaver** [comment]
Hi and thanks for contributing. 
The lib doesn't use anything from the generated code, so you don't have to generate anything. 

Just have a look how all the different endpoints are implemented. To create a new endpoint, please create a package named as in Amazon's documentation, then create a client extending Client from sp_api.base. 

A simple, working example method for post or put has a decorator, accepts **kwargs and returns 

`self._request(kwargs.pop('path'), data=kwargs)`
